File: hw/chavez/u3/hw26Observer/Python/model/stock.py
class Stock:

    def __init__(self):
        self.symbol = ""
        self.price = 0.0
        self.investors = []

    def add_observer(self, investor):
        self.investors.append(investor)

    def delete_observer(self, investor):
        self.investors.remove(investor)

    def notify_observers(self, args):
        for investor in self.investors:
            investor.update(self, args)

File: hw/chavez/u3/hw26Observer/JavaScript/model/Stock.js
class Stock {

    constructor() {
        this.symbol = "";
        this.price = 0;
        this.investors = [];
    }

    addObserver(investor) {
        this.investors.push(investor);
    }

    deleteObserver(investor) {
        this.investors = this.investors.filter(i => i !== investor);
    }

    notifyObservers(args) {
        this.investors.forEach(investor => {
            investor.update(this, args);
        });
    }

    getSymbol() {
        return this.symbol;
    }
}

export default Stock;